plugins/outputs/nats/sample.conf

```diff
@@ -31,3 +32,36 @@
   ## more about them here:
   ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
   data_format = "influx"
+
+  ## Jetstream specific configuration. If not nil, it will assume Jetstream context.
+  ## Since this is a table, it should be present at the end of the plugin section. Else you can use inline table format.
+  # [outputs.nats.jetstream]
+    ## Name of the stream, required when using jetstream. Telegraf will
+    ## use the union of the above subject and below the subjects array.
+    # name = "" 
+    # subjects = []
+    
+    ## Full jetstream create stream config, refer: https://docs.nats.io/nats-concepts/jetstream/streams
+    ## Some of the configuration fields such as `retention`, `storage`, `discard` accept integer values.
+    ## For ex- storage = 1 refers to memory(1) storage.
+
+    ## 0: limits(default), 1: interest, 2: workqueue
+    # retention = 0
+    # max_consumers = -1
+    # max_msgs_per_subject = -1
+    # max_msgs = -1
+    # max_bytes = -1
+    # max_age = 0
+    # max_msg_size = -1
+    ## 0: file(default), 1: memory
+    # storage = 0
+    ## 0: old(default), 1: new
+    # discard = 0
+    # num_replicas = 1
+    # duplicate_window = 120000000000
+    # sealed = false
+    # deny_delete = false
+    # deny_purge = false
+    # allow_rollup_hdrs = false
+    # allow_direct = true
+    # mirror_direct = false
```

Comment:
I would suggest doing
```suggestion
  ## Jetstream specific settings, if configured the plugin will assume a Jetstream context.
  # [outputs.nats.jetstream]
  #   ## Name of the stream, required when using jetstream. Telegraf will
  #   ## merge the subject above with the subjects list below.
  #   # name = "" 
  #   # subjects = []
  #   
  #   ## Full jetstream create stream config, see https://docs.nats.io/nats-concepts/jetstream/streams
  #   ## Some of the configuration fields such as `retention`, `storage`, `discard` accept integer values.
  #   ## For ex- storage = 1 refers to memory(1) storage.
  # 
  #   ## 0: limits, 1: "interest" or 2: "workqueue"
  #   # retention = 0
  #   # max_consumers = -1
  #   # max_msgs_per_subject = -1
  #   # max_msgs = -1
  #   # max_bytes = -1
  #   # max_age = 0
  #   # max_msg_size = -1
  #   ## 0: file(default), 1: memory
  #   # storage = 0
  #   ## 0: old(default), 1: new
  #   # discard = 0
  #   # num_replicas = 1
  #   # duplicate_window = 120000000000
  #   # sealed = false
  #   # deny_delete = false
  #   # deny_purge = false
  #   # allow_rollup_hdrs = false
  #   # allow_direct = true
  #   # mirror_direct = false
```
It's not clear to me what do you want to say with `#   ## For ex- storage = 1 refers to memory(1) storage.`

Furthermore, how about using strings for the enumerations `retention`, `storage` and `discard`? This would make the explanations unnecessary and will be easier to read...